dns/miekgdns2/resolver_test.go

```diff
@@ -0,0 +1,323 @@
+package miekgdns2
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+var resolvConfContents = `
+nameserver 127.0.0.1
+nameserver 127.0.0.2
+option attempts:2
```

Comment:
It'd be good to see a test that verifies that the retry behaviour is as expected (ie. that each server is attempted `attempts` times), and another that verifies what happens if `option attempts` is not present in `resolv.conf`.

Reply:
Done.